pdns/dnsdist.cc

```diff
@@ -1303,6 +1308,13 @@ try
     g_ACL.setState(acl);
   }
 
+  if (g_cmdLine.checkConfig) {
+    setupLua(true, g_cmdLine.config);
+    // No exception was thrown
+    infolog("Configuration '%s' OK!", g_cmdLine.config);
```

Comment:
Just one thing, setupLua() outputs a warning but does not throw an exception if the configuration file does not exist. I believe that's fine, but that means we will gladly validate a non-existent file, which might not be what you had in mind.